dlpar/dlpar_api/api.py

```diff
@@ -451,3 +629,552 @@ def set_virtual_proc_and_proc_units(self, linux_machine, ideal_procs,
+                                  curr_proc_units_before, self.cmd_result)
+
+    def __move_cpu_units(self, linux_machine_1, linux_machine_2, quantity):
+        """
+        Move 'quantity' proc units from linux_machine_1 to linux_machine_2.
+        """
+        # Get all values from both machines before moving
+        curr_proc_units_before_1 = float(self.get_cpu_option
+                                         (linux_machine_1, 'curr_proc_units'))
+        curr_proc_units_before_2 = float(self.get_cpu_option
+                                         (linux_machine_2, 'curr_proc_units'))
+
+        # Move the proc units
+        linux_machine = []
+        linux_machine.append(linux_machine_1)
+        linux_machine.append(linux_machine_2)
+        proc_info = []
+        proc_info.append(curr_proc_units_before_1)
+        proc_info.append(curr_proc_units_before_2)
+
+        flag = ['proc', 'm', '--procunits']
+        self.cmd_output = self.Dlpar_engine(flag, linux_machine, quantity)
+
+        self.log.debug(cmd_output)
+        self.log.debug('Sleeping for %s seconds before proceeding' %
+                       self.sleep_time)
+        time.sleep(self.sleep_time)
+
+        # Check at HMC
+        self.Dlpar_cpu_validation("cpu_m", linux_machine, quantity,
+                                  proc_info, self.cmd_output)
+
+    def __remove_cpu_units(self, linux_machine, quantity):
+        """Remove 'quantity' proc units from linux_machine."""
+        # Get all values before removing
+        curr_proc_units_before = float(self.get_cpu_option(linux_machine,
+                                                           'curr_proc_units'))
+        # Remove the cpus
+        flag = ['proc', 'r', '--procunits']
+        self.cmd_output = self.Dlpar_engine(flag, linux_machine, quantity)
+
+        self.log.debug(cmd_output)
+        self.log.debug('Sleeping for %s seconds before proceeding' %
+                       self.sleep_time)
+        time.sleep(self.sleep_time)
+
+        # Check at HMC
+        self.Dlpar_cpu_validation("cpu_r", linux_machine, quantity,
+                                  curr_proc_units_before, self.cmd_output)
+
+
+class Memory(TestCase):
+    """DLPAR memory add/move/remove test case.
+
+   The test procedure:
+    1 - Get how much of memory to test;
+    2 - Check the test environment for both linux partitions;
+    3 - Test (see run_test())
+
+    Everything is fine if we don't have troubles with the HMC and the linux
+    partitions are recognizing all added memory (using /proc/meminfo).
+    """
+
+    def __init__(self, config_payload, log='memory.log'):
+
+        TestCase.__init__(self, log, 'Memory', config_payload)
+
+        # Get test configuration
+        self.quant_to_test = 1024
+        self.quant_to_test = config_payload.get('mem_quantity_to_test')
+        self.sleep_time = 60
+        self.sleep_time = config_payload.get('sleep_time')
```

Comment:
Values are overwritten by consecutive assignments. One of the assignment should be removed.

Reply:
The above changes are addressed in the latest commit